controllers/pipelines/resource_workflow_factory.go

```diff
@@ -105,12 +106,14 @@ func (workflows *ResourceWorkflowFactory[R, ResourceDefinition]) resourceDefinit
 	return string(marshalled), nil
 }
 
-func (workflows *ResourceWorkflowFactory[R, ResourceDefinition]) ConstructCreationWorkflow(provider string, resource R) (*argo.Workflow, error) {
+func (workflows *ResourceWorkflowFactory[R, ResourceDefinition]) ConstructCreationWorkflow(provider pipelinesv1.Provider, resource R) (*argo.Workflow, error) {
 	resourceDefinition, err := workflows.resourceDefinitionYaml(resource)
 	if err != nil {
 		return nil, err
 	}
 
+	providerName := provider.ObjectMeta.Name
```

Comment:
It looks like we are doing a roundtrip from provider name, to empty provider object, to name again...without ever reading it from the cluster?

Reply:
same as above, this is a step towards loading the provider, we have separated the work into smaller PRs, the next step is to actually load the provider